test/unit/spec/signaling/v2/iceconnectionmonitor.js

```diff
@@ -0,0 +1,172 @@
+
+describe('IceConnectionMonitor', () => {
+
+  describe('.constructor', () => {
+    var pc;
+    beforeEach( () => {
+      pc = { foo: 1 };
+    });
+
+    it('stores the peerConnection provided', () => {
+      assert.equal(new IceConnectionMonitor(pc)._peerConnection, pc);
+    });
+
+    it('sets the timer to null', () => {
+      assert.equal(new IceConnectionMonitor(pc)._timer, null);
+      assert.equal(new IceConnectionMonitor(pc)._lastActivity, null);
+    });
+
+    it('defaults to 1sec of check period', () => {
+      assert.equal(new IceConnectionMonitor(pc)._activityCheckPeriodMs, 1000);
+    });
+
+    it('defaults to 3 sec of inactivity threshold', () => {
+      assert.equal(new IceConnectionMonitor(pc)._inactivityThresholdMs, 3000);
+    });
+  });
+
+  describe('.start', () => {
+    // eslint-disable-next-line no-undefined
+    ['foo', 45, { foo: 5 }, null].forEach((callback) => {
+      it(`throws if callback is ${typeof callback} `, () => {
+        const pc = { foo: 1 };
+        const monitor = new IceConnectionMonitor(pc);
+        assert.throws(() => {
+          monitor.start(callback);
+        });
+      });
+    });
+
+    it('starts the timer', () => {
+      var pc = { foo: 1 };
+      const monitor = new IceConnectionMonitor(pc);
+      assert.equal(monitor._timer, null);
+      monitor.start(() => {});
+      assert.notEqual(monitor._timer, null);
+    });
+  });
+
+  describe('._getMediaStats', () => {
+    it('extracts media stats from _getIceConnectionStats', () => {
+      var pc = { foo: 1 };
+      const iceConnectionMonitor = new IceConnectionMonitor(pc);
+      sinon.stub(iceConnectionMonitor, '_getIceConnectionStats').returns(Promise.resolve({
+        timestamp: 10,
+        otherProp: 20,
+        bytesSent: 30,
+        bytesReceived: 40
+      }));
+      return iceConnectionMonitor._getMediaStats().then((mediaStats) => {
+        assert.deepStrictEqual(mediaStats, {
+          timestamp: 10,
+          bytesSent: 30,
+          bytesReceived: 40
+        });
+      });
+    });
+
+    it('returns null if _getIceConnectionStats returns undefined', () => {
+      var pc = { foo: 1 };
+      const iceConnectionMonitor = new IceConnectionMonitor(pc);
+      // eslint-disable-next-line no-undefined
+      sinon.stub(iceConnectionMonitor, '_getIceConnectionStats').returns(Promise.resolve(undefined));
+      return iceConnectionMonitor._getMediaStats().then((mediaStats) => {
+        assert.strictEqual(mediaStats, null);
+      });
+    });
+
+    it('returns nullif _getIceConnectionStats rejects', () => {
+      var pc = { foo: 1 };
+      const iceConnectionMonitor = new IceConnectionMonitor(pc);
+      sinon.stub(iceConnectionMonitor, '_getIceConnectionStats').returns(Promise.reject(new Error('boo')));
+      return iceConnectionMonitor._getMediaStats().then((mediaStats) => {
+        assert.strictEqual(mediaStats, null);
+      });
+    });
+  });
+
+  describe('.stop', () => {
```

Comment:
@makarandp0 Nitpick: `#stop`?

Reply:
q: where do we use '.foo' v/s '#foo' ? didn't see any other usage of '#foo' in the code